api/commands/run.ts

```diff
@@ -329,6 +350,57 @@ export class Runner {
     await this.triggerChange();
     return taskResult.status;
   }
+
+  private actionHasCacheHit(action: dataform.IExecutionAction): boolean {
+    if (!this.graph.runConfig.useRunCache) {
+      return false;
+    }
+
+    if (action.type === "operation") {
+      return false;
+    }
+
+    for (const dependencyTarget of action.dependencyTargets) {
+      const dependencyAction = this.graph.actions.find(
+        action => action.target === dependencyTarget
+      );
+      if (!dependencyAction) {
+        continue;
+      }
+
+      const runResultAction = this.runResult.actions.find(
+        action => action.name === dependencyAction.name
+      );
+
+      if (
+        runResultAction &&
```

Comment:
as above: if `runResultAction` is falsey, IMO we should be blowing up, or at least returning false to this being a cache hit - case this should be impossible. i'd just remove this line altogether